### PR TITLE
[tests] Fix the most of the tests to return well defined 0 on success.

### PR DESCRIPTION
**Pull request by atamazov**
This should kill the false `make check` failures when the tests actually pass!

### CODE REVIEW

**Comment by atamazov** [comment]
@pfultz2 Well, it seems like the most recent shown config has quietly failed... How this is possible...?

**Comment by atamazov** [comment]
@asroy Maybe the reason is out of memory (Navi21)?

File: test/w_supertensor.cpp [hunk 1]
```diff
@@ -592,4 +592,8 @@ struct superTensorTest : test_driver
     }
 };
 
-int main(int argc, const char* argv[]) { test_drive<superTensorTest>(argc, argv); }
+int main(int argc, const char* argv[])
+{
+    test_drive<superTensorTest>(argc, argv);
+    return 0;
```
Comment (pfultz2) [inline test/w_supertensor.cpp:598]: I dont think this is necessary, `main()` should return `0` without needing an explicit return statement, From [cppreference](https://en.cppreference.com/w/cpp/language/main_function):

> The body of the main function does not need to contain the return statement: if control reaches the end of main without encountering a return statement, the effect is that of executing return 0;.

**Comment by pfultz2** [comment]
> Well, it seems like the most recent shown config has quietly failed... How this is possible...?

Something is calling `abort()` causing it to exit early. Adding the `return 0` no-ops wont do anything.

File: test/w_supertensor.cpp (hunk 1 above)
Comment (atamazov) [reply test/w_supertensor.cpp:598]: Never thought about this ;)

**Comment by atamazov** [comment]
@pfultz2 Yep, thanks for help))

**Pull request closed without merging**